### ISSUE TITLE
Show only changed files like git status

### ISSUE THREAD

**Issue by emirror-de** [Xuyuanp/nerdtree-git-plugin#189]
Hi there,
I did not find anything about this in the existent issues.

### Description

Having a large code base, it would be nice to switch between "only changed" and "all" file view mode during runtime, where

* "all" shows all files, as it is standard in nerdtree
* "only changed" shows only the changed files like `git status` does

### Advantage

This makes finding changed files in a large code base / folder structure much easier.

### Example

There is a code base with a complex and deep folder structure. For a new feature, two files are changed where one is in `src/........./LargeFileStructure.source` and the other one in`assets/............/LargeAssetsStructure.asset`. Between `assets` and `src` there are also a lot of folders. When vim is closed and reopened, it is required to scroll through the complete folder tree to find those two files.
If only the changed files can be displayed, it would be much faster to find and open them.

**Comment by Xuyuanp** [comment]
As I know there is no way to modify the whole tree structure.

**Comment by emirror-de** [comment]
Hm okay, that's sad. I do not have any clue about the implementation details, but it is possible to hide/show the hidden files in the tree structure with Shift + i in NerdTree. Isn't it a similar feature/behavior? Would this be an entry point of finding a way to implement this?

**Comment by Xuyuanp** [comment]
Hiding or filtering some files is an internal feature of nerdtree, not available for a plugin. I haven't read nerdtree's source code for a long time, don't know is this outdated.

**Comment by emirror-de** [comment]
According to the docs, it looks like [this could be possible](https://github.com/preservim/nerdtree/blob/81f3eaba295b3fceb2d032db57e5eae99ae480f8/doc/NERDTree.txt#L1315). I did not have time yet to have a look into the source code. But it seems that it is possible by now, if I am not wrong.

**Comment by Xuyuanp** [comment]
You're right. That's possible. We can filter out some files by calling this function ` NERDTreeAddPathFilter(callback) `. 

I've been very busy with real life stuff, will implement this feature when I have some time (about two months later, I guess).

**Comment by Xuyuanp** [comment]
ping

**Comment by emirror-de** [comment]
another ping
